File: 2021/src/test/java/nel/marco/Day8Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;

class Day8Test {

    Day8 day8 = new Day8();

    public List<String> readInput(boolean readExample) throws IOException {

        String path = "src\\main\\resources\\day8";
        if (readExample) {
            path = "src\\main\\resources\\day8Example";
        }

        List<String> stringList = Files.readAllLines(Path.of(path));

        return new ArrayList<>(stringList);

    }


    @Test
    void part1() throws IOException {

        assertEquals(26, day8.part1(readInput(true)));
        assertEquals(26, day8.part1(readInput(false)));
    }

    @Test
    void part2() throws IOException {

        assertEquals(61229, day8.part2(readInput(true)));
        assertEquals(1031553, day8.part2(readInput(false)));
    }
}

File: 2024/src/test/nel/marco/Day8Test.kt
package nel.marco

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class Day8Test {


    val day = Day8(useExample = false, useMac = false)
    val example = Day8(useExample = true, useMac = false)

    @Test
    fun answerOne() {
        assertEquals("14", example.answerOne())
        assertEquals("244", day.answerOne())
    }

    @Test
    fun answerTwo() {
        assertEquals("34", example.answerTwo())
        assertEquals("912", day.answerTwo())
    }



    fun avg() {
        val day = day
        MarcoUtil.avgTime("part1", 5) {
            assertEquals("244", day.answerOne())
        }
        MarcoUtil.avgTime("part2", 5) {
            assertEquals("912", day.answerTwo())
        }
    }
}